### PR TITLE
Heretic QoL pass

### PR DESCRIPTION
**Pull request by Absolucy**
## About The Pull Request

This makes a bunch of QoL additions and tweaks to make reworked heretics much more enjoyable.

## Why It's Good For The Game

Makes heretic less of a chore to play

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

23-09-22-1695422064-dreamseeker
23-09-22-1695422068-dreamseeker
23-09-22-1695422200-dreamseeker
23-09-22-1695422421-dreamseeker
23-09-22-1695422441-dreamseeker
23-09-22-1695422461-dreamseeker
23-09-22-1695422465-dreamseeker

</details>

## Changelog
:cl:
add: Heretics with an objective to sacrifice a head of staff can now sacrifice ANY head of staff, even non-living heart targets, if they haven't met that goal already.
tweak: Ghouls and voiceless dead now count towards the "summon X monsters from the Mansus" objective for flesh heretics.
balance: Heretic influences can now only spawn in somewhat open locations where they can be seen from a 'public' area.
add: More heretic influences will spawn midround for heretics that may be 'falling behind'.
tweak: Heretic curses now use both fingerprints and blood.
balance: Ascended heretics now draw runes MUCH faster.
balance: Rust heretics can draw runes a bit faster when in the vicinity of rusted floors.
balance: Void heretics can now draw runes a bit faster in frozen areas.
balance: Heretics now draw runes faster if they have a sacrifice target with them.
tweak: Heretic sacrifice now grabs the ghost of the victim during revival.
tweak: Heretic living heart tracking now specifies if the target is catatonic or currently on a moving shuttle.
tweak: If a heretic's sacrifice target enters cryo, it will try to roll a new sacrifice target for them.
fix: Fixed text formatting when tracking dead living heart targets.
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
needs wiki changes.

Looks good though

**Comment by Rukofamicom** [comment]
I'm not against the idea of the PR in general, but I am strictly against making it substantially easier to ascend without also changing what ascension is. Ascension needs to not grant what borders on invincibility to the heretics that achieve it.

**Comment by Absolucy** [comment]
@Rukofamicom I've removed the influence tracking - I think the new spawning system will reduce frustration compared to tracking (especially for players who are less eager to go all unga bunga in order to get someone on the funny circle)

**Comment by Rukofamicom** [comment]
I'm not against one or the other nearly as much as both together, though I like the idea of tracking moreso than moving them to only public areas I think. 

This is good enough for me if you think otherwise though.

**Comment by Absolucy** [comment]
@Rukofamicom Note that they're not _always_ in public areas, just... in the _vicinity_ of a public area (like, not in the middle of a secure area that will have sec crawling up your ass the instant you try to hack in). They can still spawn in a sec checkpoint, or a department area near a hallway (like chemistry, R&D, robotics, etc)

File: code/modules/antagonists/heretic/influences.dm [hunk 1]
```diff
@@ -64,28 +105,149 @@
...
+		/area/space,
+		/area/tcommsat,
+		/area/teleporter
+	)) + station_area_blacklist) - low_weight_areas
+
+	if(!isnum_safe(amount) || amount <= 0)
+		// 1 heretic = 4 influences
+		// 2 heretics = 7 influences
+		// 3 heretics = 9 influences
+		// 4 heretics = 10 influences, +1 for each onwards.
```
Comment (PowerfulBacon) [inline code/modules/antagonists/heretic/influences.dm:184]: Considering that they can now find these easilly and they will almost always be in public areas, this seems like an insane amount of influences to spawn. Surely with 1 heretic you almost have enough to reach ascension level on influences alone without doing any sacrafice.
I also think that if they have something that will tell them where the nearest rift is, then we don't need to check at all if the rift is spawning in a location visible to a public hallway. I would honestly prefer if they would almost always spawn somewhere that you at least have to do some hacking/tailgating to get into

**Comment by Rukofamicom** [comment]
Just as an fyi, I have fixed the broken void cloak storage in #9337

**Comment by Penwin0** [comment]
> add: More heretic influences will spawn midround for heretics that may be 'falling behind'.

I'm not sure on how wise this is

**Comment by Howluinb** [comment]
Since this is about QoL, can there be a way to delete blades remotely since having your blades taken away by sec or one guy who shoved you is basically a round-ender due to the limit on blades owned?

**Pull request closed without merging**